File: src/main/java/decorator/textreader/TextReader.java
package decorator.textreader;

import java.io.BufferedReader;
import java.io.InputStreamReader;

public abstract class TextReader {
  protected BufferedReader in=new BufferedReader(new InputStreamReader(System.in));
  void write(String[] s) {}
  void read(String[] s) {}
}

File: src/main/kotlin/decorator/textreader/TextReader.kt
package decorator.textreader

import java.io.BufferedReader
import java.io.InputStreamReader

interface TextReader {
    val `in`: BufferedReader
        get() = BufferedReader(InputStreamReader(System.`in`))

    fun write(s: Array<String>) {}
    fun read(s: Array<String>) {}
}
